third_party/move/move-model/bytecode/src/dataflow_domains.rs

```diff
@@ -310,3 +309,105 @@ impl<K: Ord + Clone, V: AbstractDomain + Clone + PartialEq> MapDomain<K, V> {
         self.extend(new_values);
     }
 }
+
+impl<A: AbstractDomain, B: AbstractDomain> AbstractDomain for (A, B) {
+    fn join(&mut self, other: &Self) -> JoinResult {
+        let mut join_res = JoinResult::Unchanged;
+        join_res = JoinResult::combine(join_res, self.0.join(&other.0));
+        join_res = JoinResult::combine(join_res, self.1.join(&other.1));
+        join_res
+    }
+}
+
+impl<T: AbstractDomain> AbstractDomain for Vec<T> {
+    fn join(&mut self, other: &Self) -> JoinResult {
+        self.iter_mut()
+            .zip(other.iter())
+            .fold(JoinResult::Unchanged, |acc, (x, y)| {
+                JoinResult::combine(acc, x.join(y))
+            })
+    }
+}
+
+/// Add a formal top and bottom element to type `T`
+#[derive(PartialEq, Eq, Clone)]
+pub enum Plus2<T> {
```

Comment:
For what is this needed? If someone would have to do this for a specific domain, it would be better done as part of the concrete Domain. A generic solutions seems to be overkill and not contribute to readability.

Also currently no use case which needs this.